apps/prairielearn/src/lib/db-types.ts

```diff
@@ -130,41 +391,94 @@ export const CourseInstanceSchema = z.object({
+
+export const CourseRequestSchema = z.object({
+  approved_by: IdSchema.nullable(),
+  approved_status: z.enum(['pending', 'approved', 'denied', 'creating', 'failed']),
+  created_at: DateFromISOString,
+  first_name: z.string().nullable(),
+  github_user: z.string().nullable(),
+  id: IdSchema,
+  institution: z.string().nullable(),
+  last_name: z.string().nullable(),
+  referral_source: z.string().nullable(),
   short_name: z.string(),
-  uid_regexp: z.string().nullable(),
-  yearly_enrollment_limit: z.number(),
+  title: z.string(),
+  user_id: IdSchema,
+  work_email: z.string().nullable(),
 });
-export type Institution = z.infer<typeof InstitutionSchema>;
+export type CourseRequest = z.infer<typeof CourseRequestSchema>;
 
-export const SamlProviderSchema = z.object({
-  certificate: z.string(),
-  email_attribute: z.string().nullable(),
+export const EnrollmentSchema = z.object({
+  course_instance_id: IdSchema,
+  created_at: DateFromISOString,
   id: IdSchema,
-  institution_id: IdSchema,
-  issuer: z.string(),
-  name_attribute: z.string().nullable(),
-  private_key: z.string(),
-  public_key: z.string(),
-  sso_login_url: z.string(),
-  uid_attribute: z.string().nullable(),
-  uin_attribute: z.string().nullable(),
-  validate_audience: z.boolean(),
-  want_assertions_signed: z.boolean(),
-  want_authn_response_signed: z.boolean(),
+  user_id: IdSchema,
 });
-export type SamlProvider = z.infer<typeof SamlProviderSchema>;
+export type Enrollment = z.infer<typeof EnrollmentSchema>;
 
-export const AuthnProviderSchema = z.object({
+export const GradingJobSchema = z.object({
+  auth_user_id: IdSchema.nullable(),
+  auto_points: z.number().nullable(),
+  correct: z.boolean().nullable(),
+  date: DateFromISOString.nullable(),
+  feedback: z.record(z.string(), z.any()).nullable(),
+  gradable: z.boolean().nullable(),
+  graded_at: DateFromISOString.nullable(),
+  graded_by: IdSchema.nullable(),
+  grading_finished_at: DateFromISOString.nullable(),
+  grading_method: z.enum(['Internal', 'External', 'Manual']).nullable(),
+  grading_received_at: DateFromISOString.nullable(),
+  grading_request_canceled_at: DateFromISOString.nullable(),
+  grading_request_canceled_by: IdSchema.nullable(),
+  grading_requested_at: DateFromISOString.nullable(),
+  grading_started_at: DateFromISOString.nullable(),
+  grading_submitted_at: DateFromISOString.nullable(),
   id: IdSchema,
-  name: z.string().nullable(),
+  manual_points: z.number().nullable(),
+  manual_rubric_grading_id: IdSchema.nullable(),
+  output: z.string().nullable(),
+  partial_scores: z.record(z.string(), z.any()).nullable(),
+  s3_bucket: z.string().nullable(),
+  s3_root_key: z.string().nullable(),
+  score: z.number().nullable(),
+  submission_id: IdSchema,
+  v2_score: z.number().nullable(),
 });
-export type AuthnProvider = z.infer<typeof AuthnProviderSchema>;
+export type GradingJob = z.infer<typeof GradingJobSchema>;
+
+export const GroupSchema = z.object({
```

Comment:
Bikeshed: I don't know if you want to clarify in a comment that you are sorting by the table/type name, not the schema name. So `GroupSchema` comes before `GroupConfigSchema` because `Group` comes before `GroupConfig`.

Reply:
I spelled that out explicitly in d88202c.